echo/index.js

```diff
@@ -254,15 +270,37 @@ function Client() {
     debug('request options', requestOptions);
 
     request.get(requestOptions, function onResp(err, response, rawBody) {
-      var statusCode = response && response.statusCode
-        ? response.statusCode
-        : 0;
-
+      var statusCode = response && response.statusCode ? response.statusCode : 0;
       if (err || parseInt(statusCode / 100) !== 2) {
-        error('[echoClient] queryAnalytics error', { err: err, body: rawBody, statusCode: statusCode });
-        callback(err || 'error response status code: ' + statusCode);
+        var errorCode;
+
+        if (err) {
+          errorCode = errorCodesAndLabels.REQUEST_ERROR;
+        } else {
+          errorCode = httpStatusToErrorCode[statusCode] === undefined
+            ? errorCodesAndLabels.UNKNOWN_ERROR
+            : httpStatusToErrorCode[statusCode];
+        }
```

Comment:
So this is duplicated from above, maybe we should create a function `errorCode = getErrorCode(err, statusCode)`